### PR TITLE
Wcdma concrete classes

### PR DESCRIPTION
**Pull request by matthewkwsiu**
This is essentially a copy-paste of the NokiaLte executor and receiver classes. Very little reviewing has been done.

### CODE REVIEW

File: src/main/java/radiounit/NokiaLteRadioUnitReceiver.java [hunk 1]
```diff
@@ -13,60 +17,130 @@
...
+		existingCarrier.setFrequencyBand(frequencyBand);
+		addCarrierInternal(existingCarrier);
 	}
 
 	public void removeCarrierNokiaLte(Integer carrierId) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[NokiaLteRadioUnitReceiver] removeCarrierNokiaLte: " + carrierId);
+		if (carriers.get(carrierId) == null) {
+			System.err.println("EricssonLteRadioUnitReceiver[] Invalid carrierId - cannot remove carrier");
```
Comment (dLeung09) [inline src/main/java/radiounit/NokiaLteRadioUnitReceiver.java]: `"NokiaLteRadioUnitReceiver"`

File: src/main/java/radiounit/NokiaLteRadioUnitReceiver.java [hunk 2]
```diff
@@ -13,60 +17,130 @@
...
 	}
 
+	/**
+	 * Private, internal method with exclusive removal access to the carriers list.
+	 * 
+	 * @param index Index of the carrier to be removed
+	 */
+	private synchronized void removeCarrierInternal(int carrierId) {
+		if (carriers.get(carrierId) == null) {
+			System.err.println("EricssonLteRadioUnitReceiver[] Invalid carrierId - cannot remove carrier");
```
Comment (dLeung09) [inline src/main/java/radiounit/NokiaLteRadioUnitReceiver.java]: `"NokiaLteRadioUnitReceiver"`

File: src/main/java/radiounit/NokiaWcdmaRadioUnitReceiver.java [hunk 3]
```diff
@@ -13,60 +17,130 @@
...
+		existingCarrier.setFrequencyBand(frequencyBand);
+		addCarrierInternal(existingCarrier);
 	}
 
 	public void removeCarrierNokiaWcdma(Integer carrierId) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[NokiaWcdmaRadioUnitReceiver] removeCarrierNokiaWcdma: " + carrierId);
+		if (carriers.get(carrierId) == null) {
+			System.err.println("EricssonWcdmaRadioUnitReceiver[] Invalid carrierId - cannot remove carrier");
```
Comment (dLeung09) [inline src/main/java/radiounit/NokiaWcdmaRadioUnitReceiver.java]: `"NokiaWcdmaRadioUnitReceiver"`

File: src/main/java/radiounit/NokiaWcdmaRadioUnitReceiver.java [hunk 4]
```diff
@@ -13,60 +17,130 @@
...
 	}
 
+	/**
+	 * Private, internal method with exclusive removal access to the carriers list.
+	 * 
+	 * @param index Index of the carrier to be removed
+	 */
+	private synchronized void removeCarrierInternal(int carrierId) {
+		if (carriers.get(carrierId) == null) {
+			System.err.println("EricssonWcdmaRadioUnitReceiver[] Invalid carrierId - cannot remove carrier");
```
Comment (dLeung09) [inline src/main/java/radiounit/NokiaWcdmaRadioUnitReceiver.java]: `"NokiaWcdmaRadioUnitReceiver"`

**Review by dLeung09 — APPROVED** [review]
Couple of typos, but looks good otherwise. Good to merge once those are fixed

**Comment by matthewkwsiu** [comment]
At first I wondered if Ctrl + F and 'replace all' hadn't worked for Strings, but then I realized the typos were in my original NokiaLte file and thus were transferred over. I'll fix those.

**Comment by matthewkwsiu** [comment]
Fixed. Thanks for catching those David

**Pull request merged**